src/main/java/org/ow2/proactive/catalog/callgraph/GraphNode.java

```diff
@@ -92,10 +112,26 @@ public String getObjectName() {
         return objectName;
     }
 
+    public String getObjectKind() {
+        return objectKind;
+    }
+
+    public boolean isInCatalog() {
+        return isInCatalog;
+    }
+
     @Override
     public String toString() {
         final StringBuilder sb = new StringBuilder();
-        sb.append(bucketName).append('/').append(objectName);
+        sb.append(" ")
+          .append(bucketName)
+          .append('/')
+          .append(objectName)
+          .append(" ")
+          .append(System.getProperty("line.separator"))
+          .append("[")
+          .append(objectKind)
+          .append("]");
```

Comment:
do you really need many appends? why not one?

Reply:
A single append is possible but for more clarity about the String structure, I used several ones.